From c8947e63d7fde16c98d49c770fc0cf3b2bea4380 Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Mon, 10 May 2010 23:10:19 +0000
Subject: [PATCH 28/28] [EVAL] Fix command -- crash

parse_command_args() returning a **argv pointer with *argv == 0 makes
dash segfault in find_command().  To reproduce run

 dash -c 'command --'

With this commit, parse_command_args() returns 0 if *argv is null after
parsing --, and so fixes the subsequent segfault.

Reported by Jonny through http://bugs.debian.org/579543

Signed-off-by: Gerrit Pape <pape@smarden.org>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

Conflicts:

	ChangeLog
---
 src/eval.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/eval.c b/src/eval.c
index d201054..5b8c241 100644
--- a/src/eval.c
+++ b/src/eval.c
@@ -652,7 +652,8 @@ parse_command_args(char **argv, const char **path)
 		if (!(c = *cp++))
 			break;
 		if (c == '-' && !*cp) {
-			argv++;
+			if (!*++argv)
+				return 0;
 			break;
 		}
 		do {
-- 
1.7.0.3

